api/src/main/java/org/eclipse/microprofile/rest/client/inject/package-info.java

```diff
@@ -19,7 +19,30 @@
  *******************************************************************************/
 
 /**
+ * APIs to aid in CDI-based injection of MP Rest Client implementations.  These
+ * annotations are used both to mark an interface as registered Rest Client and
+ * also to designate that an implementation of that interface should be injected
+ * at a specific injection point.
  *
+ * Example:
+ * <pre>
+ * @RegisterProvider
+ * @Dependent
```

Comment:
Why is `@Dependent` needed?  Can't `@RegisterProvider` be a bean-defining annotation?

Reply:
`@Dependent` is a bean-defining annotation for MP Rest Client 1.1, but this javadoc is intended for both 1.1 and 1.0 (where it is not required to be a bean-defining annotation).  We could remove `@Dependent` from the 1.1 javadocs, but that should probably be a part of PR #83.